assets/js/components/Button.js

```diff
@@ -55,6 +57,22 @@ const Button = forwardRef( ( {
 	// Use a button if disabled, even if a href is provided to ensure expected behavior.
 	const SemanticButton = ( href && ! disabled ) ? 'a' : 'button';
 
+	const getAriaLabel = () => {
+		let label = ariaLabel;
+
+		if ( target === '_blank' ) {
+			return label;
+		} else if ( typeof children === 'string' ) {
+			label = label || children;
+		}
```

Comment:
I can see how the IB wasn't 100% clear on this, but the idea was that if `target === '_blank` then the "opens in a new tab" text would be appended.

In this case, the "opens in a new tab" string will never be used because if `target === '_blank'` it returns the `ariaLabel`, which may or may not be defined.

So this should actually be:
```suggestion
		if ( target !== '_blank' ) {
			return label;
		}
		
		if ( typeof children === 'string' ) {
			label = label || children;
		}
```

(We can remove the `else` here really, as well.)

Reply:
Thanks @tofumatt 👍,
I've had a rethink on the logic here following your clarification.  I'll be interested to know if you think the changes I have made approach this better.

Reply:
The problem with that approach is it never uses the `children` when they're a `string` as part of the label. You can see that below in the snapshot, where the `aria-label` is only `"(opens in a new tab)"`, which explains _what happens when the link is opened_ but not _what the actual link is_.

So admittedly if we have a non-string `children` value in these components they need a custom aria label to work well (meaning our approach is a bit naive), but as most of the times buttons and links _do_ have `children` or `label` props that are strings it works well.

In the case of the test below whose snapshot you updated, what you want is `aria-label="Let’s go (opens in a new tab)"`. `aria-label` is used to essentially _replace_ what a screenreader would normally read out as the contents of the link.

Imagine if you had a close button like: `<a href="#close">X</a>`. A screenreader would read that link as "X". Not very accessible/usable, but doing `<a href="#close" aria-label="Close window">X</a>` would have a screenreader read that link as "Close window" and NOT read out the "X".

So that's what we want here; your approach won't quite work for that, it'd need to be something like:

```js
	const getAriaLabel = () => {
		let label = ariaLabel;

		if ( target !== '_blank' ) {
			return label;
		}

		const newTabText = _x( '(opens in a new tab)', 'screen reader text', 'google-site-kit' );

		if ( typeof children === 'string' ) {
			label = label || children;
		}

		if ( label ) {
			return `${ label } ${ newTabText }`;
		}

		return newTabText;
	};
```

Reply:
Many thanks @tofumatt 👍